accounts_accept_terms: Make elements looks similar to other pages.

amanagr opened the pull request:
Add even vertical space between elements.
<img width="1060" alt="Screenshot 2021-12-19 at 7 53 52 PM" src="https://user-images.githubusercontent.com/25124304/146679041-e70b864d-49d1-450f-a412-ee7d2250ca00.png">

timabbott commented:
Is "Continue to login" the right button title?  "Enter" is worse, but I think we want something more like "Continue to Zulip", since you get this after completing the login process.

@alya thoughts?

alya commented:
The standard labels for the button seem to be either "Accept" or "I agree".

alya commented:
I feel like we need to provide a bit more info here. Maybe something like:
---
## New Terms of Service

We have been managing chat.zulip.org as though the Zulip Terms of Service and Privacy Policy apply to it for some time. To make this de facto policy official, we are asking all users to acknowledge the recently updated Zulip Terms of Service.

[] I agree to the Terms of Service.

[Accept]

----

Will this be shown only to existing users? The above language wouldn't make sense for new users.

amanagr commented:
<img width="1195" alt="Screenshot 2022-01-02 at 10 52 29 AM" src="https://user-images.githubusercontent.com/25124304/147867032-ee30843e-9c57-4a1e-b368-8511aafaec10.png">

Majority of people seeing this will be existing users at first and then new users.

timabbott commented:
I think this needs a few changes:
* We should make the chat.zulip.org detail a setting -- this UI widget is also used for organizations migrated to Zulip Cloud from self-hosting or Slack, where that'd be inappropriate. Simplest might be if we can just put the HTML for that additional notice paragraph in a setting in `/etc/zulip/settings.py`, like we do for the homepage banner (`LANDING_PAGE_NAVBAR_MESSAGE`).
* The text needs some tweaks, but that can be independent once we do the above. (Specifically, the current text would be confusing to read for users seeing it in 2023, and so we should write it to refer to the current date rather than "recently")

amanagr commented on zproject/dev_settings.py:
```diff
@@ -166,6 +166,7 @@
 
 BILLING_ENABLED = True
 LANDING_PAGE_NAVBAR_MESSAGE: Optional[str] = None
+TERMS_OF_SERVICE_MESSAGE: Optional[str] = "We have been managing <a href='https://www.chat.zulip.org'>chat.zulip.org</a> as though the Zulip Terms of Service and Privacy Policy apply to it for some time. To make this de facto policy official, we are asking all users to acknowledge the recently updated Zulip Terms of Service."
```
I left the old message for you to modify / test.

timabbott commented:
Pushed back here after changing the `dev_settings` sample prompt and moving where these sit in the settings files to be with the other ToS stuff.

It looks like we have a pre-existing `FIRST_TIME_TERMS_OF_SERVICE_TEMPLATE` setting with a similar purpose. I think it's potentially OK to have both -- the first-time one is just for folks whose current ToS state is `None`, and the second is for a new ToS version -- they're clearly different situations.  But we should clean up a few things.
* Probably the notices should appear in the same part of the page and be styled more or less identically?
* For context, `templates/corporate/zulipchat_migration_tos.html` is what folks see when first logging in after a migration from Slack or whatnot: 
```
<div class="pitch">
    {% trans %}
    <p>This organization has been migrated to new hosting on zulip.com. We just need you to do one last thing.</p>
    <h3>Accept the Zulip Cloud terms of service</h3>
    {% endtrans %}
</div>
```

@amanagr can you make a screenshot for what it looks like with that set and try to make the CSS support both nicely?  
```
FIRST_TIME_TERMS_OF_SERVICE_TEMPLATE = "corporate/zulipchat_migration_tos.html"
```

@alya thoughts?

alya commented:
I can comment once there's a screenshot to look at.

amanagr commented:
![Screenshot 2022-01-25 at 7 20 54 PM](https://user-images.githubusercontent.com/25124304/150989266-b3e9211c-b0f6-46a2-a762-8effd2e941ca.png)

I removed `new` world from the headline to generalize it. Moved the `special_message_template` position to be same as `terms_of_service_message`.

alya commented:
OK, how about something like this? If I understand correctly, this can be shown when an organization migrates from self-hosted Zulip or Slack, etc. -- is that right?

> This team chat is now being hosted on Zulip Cloud. Please accept the [Zulip Terms of Service](https://zulip.com/policies/terms) to continue.

amanagr commented:
Sure, sounds better.
![Screenshot 2022-01-27 at 12 05 17 PM](https://user-images.githubusercontent.com/25124304/151304814-f2b31657-b1ef-499b-8c2d-926a62d9f989.png)

timabbott commented:
The two parallel settings were still pretty confusing, especially with the code for both of them in different places.  I adding comments for them in `default_settings.py` and pushed an added commit that should make the implementation easier to follow.  @amanagr can you test?

I think it would actually be worth a bit more fiddling to rename `terms_of_service_message_template` to `first_time_terms_of_service_message_template` in the `context` and template, just to make the two variables not sound almost identical.

amanagr commented:
Tested the changes and renamed the variable to `first_time_terms_of_service_message_template ` too.

timabbott commented:
Merged, thanks @amanagr!  Sorry for the delay in doing a final review on this.
